src/NodaTime.Test/DateIntervalTest.cs

```diff
@@ -208,5 +218,95 @@ public void Deconstruction()
+        [TestCase("2014-03-07,2014-03-31", "2014-03-07,2014-03-15", true)]
+        [TestCase("2014-03-07,2014-03-31", "2014-03-10,2014-03-31", true)]
+        [TestCase("2014-03-07,2014-03-31", "2014-03-10,2014-03-15", true)]
+        [TestCase("2014-03-07,2014-03-31", "2014-03-05,2014-03-09", false)]
+        [TestCase("2014-03-07,2014-03-31", "2014-03-20,2014-04-07", false)]
+        [TestCase("2014-11-01,2014-11-30", "2014-01-01,2014-12-31", false)]
+        public void Contains_IntervalOverload(string firstInterval, string secondInterval, bool expectedResult)
+        {
+            DateInterval value = ParseInterval(firstInterval);
+            DateInterval other = ParseInterval(secondInterval);
+            Assert.AreEqual(expectedResult, value.Contains(other));
+        }
+
+        [Test]
+        public void Intersection_NullInterval_Throws()
+        {
+            var value = new DateInterval(new LocalDate(100), new LocalDate(200));
+            Assert.Throws<ArgumentNullException>(() => value.Intersection(null));
+        }
+
+        [Test]
+        public void Intersection_IntervalInDifferentCalendar_Throws()
+        {
+            var value = new DateInterval(
+                new LocalDate(2017, 11, 6, CalendarSystem.Gregorian),
+                new LocalDate(2017, 11, 10, CalendarSystem.Gregorian));
+
+            var other = new DateInterval(
+                new LocalDate(2017, 11, 6, CalendarSystem.Coptic),
+                new LocalDate(2017, 11, 10, CalendarSystem.Coptic));
+
+            Assert.Throws<ArgumentException>(() => value.Intersection(other));
+        }
+
+        [TestCase("2014-03-07,2014-03-07", "2014-03-07,2014-03-07", "2014-03-07,2014-03-07")]
+        [TestCase("2014-03-07,2014-03-10", "2015-01-01,2015-04-01", null)]
+        [TestCase("2015-01-01,2015-04-01", "2014-03-07,2014-03-10", null)]
+        [TestCase("2014-03-07,2014-03-31", "2014-03-07,2014-03-15", "2014-03-07,2014-03-15")]
+        [TestCase("2014-03-07,2014-03-31", "2014-03-10,2014-03-31", "2014-03-10,2014-03-31")]
+        [TestCase("2014-03-07,2014-03-31", "2014-03-10,2014-03-15", "2014-03-10,2014-03-15")]
+        [TestCase("2014-03-07,2014-03-31", "2014-03-05,2014-03-09", "2014-03-07,2014-03-09")]
+        [TestCase("2014-03-07,2014-03-31", "2014-03-20,2014-04-07", "2014-03-20,2014-03-31")]
+        [TestCase("2014-11-01,2014-11-30", "2014-01-01,2014-12-31", "2014-11-01,2014-11-30")]
+        public void Intersection(string firstInterval, string secondInterval, string expectedInterval)
+        {
+            var value = ParseInterval(firstInterval);
+            var other = ParseInterval(secondInterval);
+            var expectedResult = ParseInterval(expectedInterval);
+            Assert.AreEqual(expectedResult, value.Intersection(other));
+        }
+        
+        private DateInterval ParseInterval(string textualInterval)
+        {
+            if (textualInterval == null)
```

Comment:
I'll fix this up myself after merging - I don't want to keep asking for tiny things :)